python/tk_multi_workfiles/work_area.py

```diff
@@ -18,34 +18,48 @@
 from .user_cache import g_user_cache
 from .util import Threaded
 
+
 class WorkArea(object):
     """
     Class containing information about the current work area including context, templates
     and other miscelaneous work-area specific settings.
     """
-    
+
     class _SettingsCache(Threaded):
         """
-        Cache of settings per context, engine and app name.
+        Cache of settings per context.
         """
+
         def __init__(self):
+            """
+            Constructor.
+            """
             Threaded.__init__(self)
-            self._cache = {}
+            self._cache = []
 
         @Threaded.exclusive
-        def get(self, engine_name, app_name, context):
```

Comment:
We're not taking settings from workfiles v1 anymore, nor are we taking settings from other engine instances, so those values are not necessary for computing a key anymore.